### PR TITLE
Feature/340 cv search found in

### PR DESCRIPTION
**Pull request by DominikFischli**
Implemented issue #340 

For now, the found-in tag has been implemented to look like a button and links to the page of the person who has been found.
This should facilitate the implementation of #341

### CODE REVIEW

**Comment by psunix** [comment]
please also make sure not creating n+1 queries. maybe you already addressed this ... just came up my mind ;)
https://semaphoreci.com/blog/2017/08/09/faster-rails-eliminating-n-plus-one-queries.html

**Comment by DominikFischli** [comment]
Changes Implemented:

* Refactored the search to domain/PeopleSearch
* Didn't use Serializer or Models (Frontend & Backend), instead, render the results from PeopleSearch directly as JSON
* The endpoint api/people/search?q=Peter returns the person id, the person name and where the person contains the query in the format:
``
{
  person: { id: 288, name: Hans Peter }
  found_in: Name
}, ...
``
* In the frontend routes/cv-search.js fetches api/people/search?q=...

**Comment by DominikFischli** [comment]
Requested changes have been addressed, thanks for the Feedback!

**Comment by DominikFischli** [comment]
Changes have been addressed, thanks for the feedback!

File: app/domain/people_search.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    res = in_attributes(person.attributes)
+    res ||= in_associations(person)
+    res.try(:camelize, :lower)
+  end
+
+  # Load the attributes of the given people into cache
+  # Without this, reflective methods accessing attributes over associations
+  # would come up empty
+  def pre_load(people)
+    person_keys = people.map { |person| person.id }
```
Comment (TheWalkingLeek) [inline app/domain/people_search.rb:34]: Sorry I forgot about this in my last review.
If you only call one method on each iteration object you can just use `people.map(&:id)`
That should do the trick

File: app/domain/people_search.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                      association: sym.to_s, attribute_name: attribute_name)
+      end
+    end
+    nil
+  end
+
+  def association_symbols
+    Person.reflections.keys.map do |key|
+      key.to_sym
+    end
```
Comment (TheWalkingLeek) [inline app/domain/people_search.rb:55]: Same here

**Comment by DominikFischli** [comment]
Done

**Pull request merged**